urth/dashboard/converter/static/urth/dashboard.js

```diff
@@ -17,53 +25,46 @@ define(['jquery', 'Thebe', 'urth-common/error-log'], function($, Thebe, ErrorLog
         return null;
     }
 
-    function initThebe(args) {
-        thebe = new Thebe.Thebe({
-            url: args.url,
-            debug: false,
-            append_kernel_controls_to: 'body',
-            next_cell_shortcut: false,
-            inject_css: false, // don't pull in mathjax, jqueryui, etc. for now
-            tmpnb_mode: Urth.tmpnb_mode || false,
-            kernel_name: Urth.kernel_name,
-            load_css: false
-        });
-
-        window.IPython.notebook = thebe.notebook;
-        //  A Hack for exposing the base_url to the urth widgets
-        thebe.notebook.base_url = '/';
-
-        // show a busy indicator when communicating with kernel
-        var debounced;
-        thebe.events.on('kernel_busy.Kernel kernel_idle.Kernel', function(event) {
-            clearTimeout(debounced);
-            debounced = setTimeout(function() {
-                var isBusy = event.type === 'kernel_busy';
-                $('.busy-indicator')
-                    .toggleClass('show', isBusy)
-                    // Prevent progress animation when hidden by removing 'active' class.
-                    .find('.progress-bar')
-                        .toggleClass('active', isBusy);
-            }, 500);
-        });
+    function initKernel(args) {
+        var kernelReady = $.Deferred();
+
+        // TODO
+        // 1. Figure out how to load `display_data` comm messages (includes source for matplotlib, styling, etc)
+        // 2. Create this global, so extensions like matplotlib can register a comm callback:
+        //      IPython.notebook.kernel.comm_manager.register_target('matplotlib', mpl.mpl_figure_comm)
+
+        var options = {
+            baseUrl: 'http://192.168.99.100:9500',
```

Comment:
Reminder for later: read Urth options instead of hardcoding. Also, need to mimic spawn call for container that thebe does ahead of requesting a kernel. (jupyter-js-services assumes the kernel provider container is pre-running, i.e., the notebook server.)